packages/medusa/src/strategies/product-export.ts

```diff
@@ -0,0 +1,662 @@
+    ],
+    [
+      "Variant Material",
+      {
+        accessor: (variant: ProductVariant): string =>
+          variant?.material?.toString() ?? "",
+        entityName: "variant",
+      },
+    ],
+  ])
+
+  private readonly NEWLINE_ = "\r\n"
+  private readonly DELIMITER_ = ";"
+  private readonly BATCH_SIZE = 50
+
+  constructor({
+    manager,
+    batchJobService,
+    productService,
+    fileService,
+    productRepository,
+  }: InjectedDependencies) {
+    super({
+      manager,
+      batchJobService,
+      productService,
+      fileService,
+      productRepository,
+    })
+
+    this.manager_ = manager
+    this.batchJobService_ = batchJobService
+    this.productService_ = productService
+    this.fileService_ = fileService
+    this.productRepository_ = productRepository
+  }
+
+  async buildTemplate(): Promise<string> {
+    return this.buildHeader()
+  }
+
+  async prepareBatchJobForProcessing(
+    batchJob: AdminPostBatchesReq,
+    req: Request
+  ): Promise<AdminPostBatchesReq> {
+    const { limit, offset, order, fields, expand, ...context } = batchJob.context
+
+    const { select: filterableFields, skip, take, ...listConfig } = prepareListQuery(
+      {
+        context: batchJob.context,
+        type: batchJob.type,
+        dry_run: batchJob.dry_run,
+        limit: limit as number,
+        offset: offset  as number,
+        order: order as string,
+        fields: fields as string,
+        expand: expand as string,
+      },
+      {
+        isList: true,
+        defaultRelations: this.relations_
+      },
+    )
+
+    batchJob.context = {
+      ...(context ?? {}),
+      listConfig,
+      filterableFields,
+      offset: skip,
+    }
+
+    return batchJob
+  }
+
+  async processJob(batchJobId: string): Promise<void> {
+    let offset = 0
+    let advancementCount = 0
+    let productCount = 0
+
+    return await this.atomicPhase_(
+      async (transactionManager) => {
+        let batchJob = await this.batchJobService_
+          .withTransaction(transactionManager)
+          .retrieve(batchJobId)
+
+        const {
+          writeStream,
+          url: key,
+          promise,
+        } = await this.fileService_.getUploadStreamDescriptor({
+          name: "product-export",
+          ext: "csv",
+        })
+
+        const header = await this.buildHeader()
+        writeStream.write(header)
+
+        offset = batchJob.context.offset ?? offset
+        advancementCount = batchJob.result.advancementCount ?? advancementCount
+        const { listConfig, filterableFields } = batchJob.context as Context
+
+        const [productList, count] = await this.productService_
+          .withTransaction(transactionManager)
+          .listAndCount(filterableFields, {
+            ...(listConfig ?? {}),
+            skip: offset,
+            take: this.BATCH_SIZE,
+          })
+
+        productCount = count
+        let products = productList
+
+        while (advancementCount < productCount) {
```

Comment:
I think this should be `offset < productCount` because the batchjob can be created with an offset and this would cause errors when rows are counted beyond the productCount

Reply:
👍